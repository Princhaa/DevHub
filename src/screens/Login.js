/* @flow */

import React, { Component } from 'react'
import { View, Text, StyleSheet, ActivityIndicator } from 'react-native'

import CustomButton from '../components/CustomButton'
import CustomTextInput from '../components/CustomTextInput'

import constants from '../config/constants'

export default class Login extends Component {
  static navigationOptions = {
    header: null
  }

  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      isLoggingIn: false
    }

    this._handleSubmitEditing = this._handleSubmitEditing.bind(this)
    this._handleLoginPress = this._handleLoginPress.bind(this)
  }

  _handleTextInputChange(stateChange) {
    this.setState(stateChange)
  }

  _handleSubmitEditing(nextRef) {
    if (typeof nextRef === 'string') {
      this._handleLoginPress()
      return
    }
    nextRef.focus()
  }

  _handleLoginPress() {
    this.setState({ isLoggingIn: !this.state.isLoggingIn })
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>Welcome to</Text>
        <Text style={styles.title}>DevHub</Text>
        <CustomTextInput
          ref={ref => (this.emailRef = ref)}
          placeholder={'Email'}
          keyboardType={'email-address'}
          autoCapitalize={'none'}
          returnKeyType={'next'}
          onChangeText={value => this._handleTextInputChange({ email: value })}
          onSubmitEditing={this._handleSubmitEditing.bind(this, this.passwordRef)}
        />
        <CustomTextInput
          ref={ref => (this.passwordRef = ref)}
          placeholder={'Password'}
          secureTextEntry={true}
          autoCapitalize={'none'}
          returnKeyType={'done'}
          onChangeText={value => this._handleTextInputChange({ password: value })}
          onSubmitEditing={this._handleSubmitEditing.bind(this, 'done')}
        />
        <CustomButton color={constants.PRIMARY_COLOR} onPress={() => this._handleLoginPress()}>
          {this.state.isLoggingIn ? (
            <ActivityIndicator color={'white'}/>
          ) : (
            <Text style={{ color: 'white' }}>Login</Text>
          )}
        </CustomButton>
        <Text style={styles.caption}>Register here</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    justifyContent: 'center',
    backgroundColor: 'white'
  },

  title: {
    fontSize: 30,
    fontWeight: 'bold',
    marginVertical: 10
  },

  caption: {
    position: 'absolute',
    bottom: 30,
    alignSelf: 'center'
  }
})
