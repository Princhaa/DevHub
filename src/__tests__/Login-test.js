import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Login from '../screens/Login'
import { shallow } from 'enzyme'

it('Renders Login Correctly', () => {
  const renderer = new ShallowRenderer()
  renderer.render(<Login />)
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})

describe('Testing Login button to change state', () => {
  it('State changes as expected', () => {
    const wrapper = shallow(<Login />)
    expect(wrapper).toMatchSnapshot()
    wrapper.instance()._handleLoginPress()
    expect(wrapper.state().isLoggingIn).toEqual(true)
  })
})
