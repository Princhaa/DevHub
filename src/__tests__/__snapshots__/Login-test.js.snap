// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Login Correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Welcome to
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
        "marginVertical": 10,
      }
    }
  >
    DevHub
  </Text>
  <CustomTextInput
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Email"
    returnKeyType="next"
  />
  <CustomTextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Password"
    returnKeyType="done"
    secureTextEntry={true}
  />
  <CustomButton
    color="#008BA3"
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </CustomButton>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "bottom": 30,
        "position": "absolute",
      }
    }
  >
    Register here
  </Text>
</Component>
`;

exports[`Testing Login button to change state State changes as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Welcome to
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
          DevHub
        </Text>,
        <CustomTextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          returnKeyType="done"
          secureTextEntry={true}
        />,
        <CustomButton
          color="#008BA3"
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Login
          </Text>
        </CustomButton>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "bottom": 30,
              "position": "absolute",
            }
          }
        >
          Register here
        </Text>,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Welcome to",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Welcome to",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "DevHub",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": "DevHub",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "keyboardType": "email-address",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Email",
          "returnKeyType": "next",
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Password",
          "returnKeyType": "done",
          "secureTextEntry": true,
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Login
          </Text>,
          "color": "#008BA3",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Login",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Register here",
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "center",
            "bottom": 30,
            "position": "absolute",
          },
        },
        "ref": null,
        "rendered": "Register here",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Welcome to
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 10,
              }
            }
          >
            DevHub
          </Text>,
          <CustomTextInput
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            returnKeyType="done"
            secureTextEntry={true}
          />,
          <CustomButton
            color="#008BA3"
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Login
            </Text>
          </CustomButton>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "bottom": 30,
                "position": "absolute",
              }
            }
          >
            Register here
          </Text>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Welcome to",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Welcome to",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "DevHub",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": "DevHub",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Email",
            "returnKeyType": "next",
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Password",
            "returnKeyType": "done",
            "secureTextEntry": true,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Login
            </Text>,
            "color": "#008BA3",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Login",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Register here",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "bottom": 30,
              "position": "absolute",
            },
          },
          "ref": null,
          "rendered": "Register here",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
